aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1079,6 +1079,12 @@ Tensor trace_cpu(const Tensor& self) {
   return result;
 }
 
+// TODO: this routine should be implemented without diag and sum for perf problems,
```

Comment:
File a new issue to improve the performance of linalg.trace and point to that issue in this comment

Reply:
Actually (see below) I recommend we address this in this PR and not file a follow-up issue.